logstash-core/lib/logstash/patches/clamp.rb

```diff
@@ -0,0 +1,47 @@
+require 'clamp'
+require 'logstash/environment'
+
+module Clamp
+  module Attribute
+    class Instance
+      def default_from_environment
+        # we don't want uncontrolled var injection from the environment
```

Comment:
The reason is simply a matter of design, I built this PR so settings can be retrieved from the cli, then yaml file, then defaults. Also allowing env would make it a bit too complex, IMO.

Adding an extra comment describing this.